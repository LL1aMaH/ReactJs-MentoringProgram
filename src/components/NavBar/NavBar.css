.navbar {
  display: flex;
  justify-content: space-between;
  padding: 1rem 5rem;
  margin-top: 1rem;
  background-color: var(--secondary-background-color);
  align-items: center;
}

.buttons {
  display: flex;
}

.dropdown {
  max-width: 12rem;
}

.button {
  margin-right: 1rem;
  font-size: 17px;
}

.selectOption {
  background-color: var(--primary-link-color);
}

.label {
  cursor: default;
}
