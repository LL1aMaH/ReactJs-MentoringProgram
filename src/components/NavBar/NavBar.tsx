import React, { memo, SyntheticEvent } from 'react';

import { Button } from 'Components/Button';

import { NAV_BAR_BUTTONS, SORT_BY } from 'Consts/index';
import { SelectField } from 'Components/NavBar/components/SelectField';

import classnames from 'classnames';

import styles from './NavBar.css';

export type TNavBarProps = {
  activeButton: string;
  handleClick: (e: SyntheticEvent) => void;
  handleSelect: (value: string) => void;
};

export const NavBar = memo(function NavBar({
  activeButton,
  handleClick,
  handleSelect,
}: TNavBarProps): JSX.Element {
  return (
    <div className={styles.navbar}>
      <div className={styles.buttons}>
        {NAV_BAR_BUTTONS.map((button) => (
          <Button
            key={button}
            className={classnames(styles.button, {
              [styles.activeButton]: button === activeButton,
            })}
            onClick={handleClick}
          >
            {button}
          </Button>
        ))}
      </div>
      <SelectField
        options={SORT_BY}
        onChange={handleSelect}
        className={styles.dropdown}
        buttonClassName={styles.select}
        optionClassName={styles.select}
        label="SORT BY"
        selectedOptionClassName={styles.selectOption}
        labelClassName={styles.label}
      />
    </div>
  );
});
