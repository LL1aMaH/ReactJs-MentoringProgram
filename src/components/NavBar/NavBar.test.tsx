import React from 'react';
import { render } from '@testing-library/react';

import { NavBar } from './NavBar';

describe('NavBar', () => {
  it('renders correctly', () => {
    const component = render(
      <NavBar activeButton="ALL" handleClick={jest.fn} handleSelect={jest.fn} />,
    );
    expect(component.asFragment()).toMatchSnapshot();
  });
});
