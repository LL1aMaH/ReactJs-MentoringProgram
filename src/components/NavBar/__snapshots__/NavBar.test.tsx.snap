// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="buttons"
    >
      <button
        class="button activeButton button"
        tabindex="0"
      >
        ALL
      </button>
      <button
        class="button button"
        tabindex="0"
      >
        DOCUMENTARY
      </button>
      <button
        class="button button"
        tabindex="0"
      >
        COMEDY
      </button>
      <button
        class="button button"
        tabindex="0"
      >
        HORROR
      </button>
      <button
        class="button button"
        tabindex="0"
      >
        CRIME
      </button>
    </div>
    <div
      class="dropDown"
    >
      <span
        class="label"
      >
        SORT BY
      </span>
      <div
        class="container dropdown"
      >
        <button
          class="selectButton select dropdownClose button"
          data-testid="selectButton"
          tabindex="0"
        >
          RELEASE DATE
          <svg />
        </button>
        <ul
          class="selectList close"
        >
          <li>
            <button
              class="option select selectOption"
              type="button"
            >
              RELEASE DATE
            </button>
          </li>
          <li>
            <button
              class="option select"
              type="button"
            >
              RATING
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
